bin/tic-tac-toe.rb

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env ruby
+require '../lib/board.rb'
+require '../lib/player.rb'
+
+puts "\n"
+puts "**************************************************"
+puts "*                                                *"
+puts "*       Let's Play                               *"
+puts "*   _______    _______      _______              *"
+puts "*  |__   __|  |__   __|    |__   __|        __   *"
+puts "*     | |    _   | |  _   _   | |  _   _   | /   *"
+puts "*     | | o |    | |  _| |    | | | | |_|  |/    *"
+puts "*     |_| | |_   |_| |_| |_   |_| |_| |_   o     *"
+puts "*                                                *"
+puts "**************************************************"
+puts "\n"
+puts "Welcome! On each turn, the player must choose a number"
+puts "corresponding to the board tile he wants to mark."
+puts "The first player to place three marks in a straight line"
+puts "wins the match. You can play as many matches as you want."
+puts "The final score will be shown at the end of the game."
+
+userX = Player.new("X")
+userO = Player.new("O")
+active_player = rand()*10 < 5 ? userO : userX #this randomly chooses the starting player
+
+play_again = true
+while play_again
```

Comment:
I think tic-tac-toe.rb is doing a little too much in itself here. Without reading too much of the details of lines 28-98 I believe it's the whole game cycle right?

In this case here are things needed to be worked on:

1. You should have a Game class which takes in as parameters two player objects and the board object
2. The Game class' responsibility is mainly to manage the flow of the game (by making sure the players make turns alternatively, and also calling the appropriate instance methods for each objects it has as parameters [player1, player2 and board], and likewise checking if there's already a winner or if it's a draw/tie).
3. The main tic-tac-toe.rb file should mainly instantiate all the objects needed for this game: userX, userO, board and game and then just call the appropriate game method (probably a game_cycle method) in order to run the game.